autopilot/wallet/wallet.go

```diff
@@ -0,0 +1,107 @@
+package wallet
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.sia.tech/core/types"
+	cwallet "go.sia.tech/coreutils/wallet"
+	"go.sia.tech/renterd/alerts"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/autopilot/contractor"
+	"go.uber.org/zap"
+)
+
+type (
+	Bus interface {
+		Wallet(ctx context.Context) (api.WalletResponse, error)
+		WalletPending(ctx context.Context) (resp []cwallet.Event, err error)
+		WalletRedistribute(ctx context.Context, outputs int, amount types.Currency) (ids []types.TransactionID, err error)
+	}
+
+	Wallet interface {
+		PerformWalletMaintenance(ctx context.Context, cfg api.AutopilotConfig) error
+	}
+)
+
+type (
+	wallet struct {
```

Comment:
Not sure `wallet` is the best name for this. Maybe `WalletMaintainer` is slightly more accurate.